### ISSUE TITLE
Catching exceptions [Future resolved more than once]

### ISSUE THREAD

**Issue by jaskinn** [peerlibrary/meteor-aws-sdk#12]
I am getting a AWS SQS Queue name (in response to a POST from an external source) - no problems. So I moved on to testing what happens when I cannot.

The first error appears to drop through to the catch as you would expect, then the second time also falls through to the catch - however then Meteor exits (code 8) with 'Error: Future resolved more than once'.

```javascript
Router.route('/notify', {
    name: 'notify',
    where: 'server'
    })
    .post(function() {
        console.log('starting notify');
        try {
            var sqs = new AWS.SQS();
            var queueParams = {
                QueueName: 'nup',
                QueueOwnerAWSAccountId: '42'
            };
            var res = sqs.getQueueUrlSync(queueParams);
         }
         catch(e) {
            console.log(e);
         }
        this.response.writeHead(200, {'Content-Type': 'text/html'});
        this.response.end();
    }
);
```

**Comment by mitar** [comment]
Not sure what is happening here. The tricky thing in fiber is that there are two ways to convey the error. One is throwing an error, another is calling the callback from async function with first argument non-null. What could be happening is that both of those are triggered in your case.

If you manage to fix it, pull requests are welcome.

**Comment by mpranjic** [comment]
It does not only happen on error. It happens whenever you run it twice (easy way to reproduce is ```meteor shell```). At least for me, it doesn't matter if AWS returns error or not.
@jaskinn are you sure that it works without issues when there is no exception thrown?
I have defined two functions:
1. runSuccess();
2. runFail();
Obviously first one should pass, latter should raise exception.
What happens for me is that if I call ```runSuccess(); runFail();``` then both pass. If I run ```runFail(); runSuccess();``` then both fail. So the second one always does the same as the first one. In all cases I get ```Future resolved more than once``` after second call.

btw, I am using AWS.S3, but it's exactly the same issue.

**Comment by mitar** [comment]
A month ago I reimplemented [blocking](https://github.com/peerlibrary/meteor-blocking). Can you maybe force old version of it and see if it works then?

**Comment by mpranjic** [comment]
I have tried with ```Meteor.wrapAsync``` instead of using "Sync" method and nothing changes.

**Comment by jaskinn** [comment]
I'm not completely certain it runs without issues (without an exception thrown). I was unable to puzzle out what was causing the issue, so I am using AWS via meteorhacks:async - everything is working as expected.

**Comment by jaskinn** [comment]
I also tried `Meteor.wrapAsync` without any success.

**Comment by mitar** [comment]
Does anyone want to make a pull request which would test this? Is there an open S3 or some other API we could test against?

**Comment by mpranjic** [comment]
My bad, I wrapped the wrong part. So it works properly with ```Meteor.wrapAsync``` for me.

**Comment by mitar** [comment]
Can you force old version of blocking? And make a test for this?

**Comment by mpranjic** [comment]
Sure, I will.

**Comment by mitar** [comment]
I released a bugfix version `peerlibrary:aws-sdk@2.1.14_1`. Thank you for reporting this and sorry for any issues you might have because of this.
